parser/parser.go

```diff
@@ -379,133 +388,88 @@ func (p *parser) evaluateOperator(value1, value2 Expression, operator rune,
-
-	inMap1 := make(map[string]*Property)
-	inMap2 := make(map[string]*Property)
-	inBoth := make(map[string]*Property)
-
-	for _, prop1 := range map1 {
-		inMap1[prop1.Name] = prop1
-	}
-
-	for _, prop2 := range map2 {
-		inMap2[prop2.Name] = prop2
-		if _, ok := inMap1[prop2.Name]; ok {
-			inBoth[prop2.Name] = prop2
-		}
-	}
-
-	for _, prop1 := range map1 {
-		if prop2, ok := inBoth[prop1.Name]; ok {
-			var err error
-			newProp := *prop1
-			newProp.Value, err = p.evaluateOperator(prop1.Value, prop2.Value, '+', pos)
-			if err != nil {
-				return nil, err
-			}
-			ret = append(ret, &newProp)
-		} else {
-			ret = append(ret, prop1)
-		}
-	}
-
-	for _, prop2 := range map2 {
-		if _, ok := inBoth[prop2.Name]; !ok {
-			ret = append(ret, prop2)
-		}
-	}
-
-	return ret, nil
-}
-
-func (p *parser) parseOperator(value1 Expression) *Operator {
-	operator := p.tok
-	pos := p.scanner.Position
-	p.accept(operator)
-
-	value2 := p.parseExpression()
-
-	value, err := p.evaluateOperator(value1, value2, operator, pos)
-	if err != nil {
-		p.error(err)
-		return nil
-	}
-
-	return value
-
-}
-
 func (p *parser) parseValue() (value Expression) {
 	switch p.tok {
 	case scanner.Ident:
-		return p.parseVariable()
+		switch text := p.scanner.TokenText(); text {
+		case "true", "false":
+			value = &Bool{
+				LiteralPos: p.scanner.Position,
+				Value:      text == "true",
+			}
+		default:
+			if p.eval {
+				if assignment, local := p.scope.Get(text); assignment == nil {
+					p.errorf("variable %q is not set", text)
+				} else {
+					if local {
+						assignment.Referenced = true
+					}
+					value = assignment.Value
+				}
+			}
+			value = &Variable{
+				Name:    text,
+				NamePos: p.scanner.Position,
+				Value:   value,
+			}
+		}
+		p.accept(scanner.Ident)
 	case scanner.String:
-		return p.parseStringValue()
-	case '[':
-		return p.parseListValue()
-	case '{':
-		return p.parseMapValue()
-	default:
-		p.errorf("expected bool, list, or string value; found %s",
-			scanner.TokenString(p.tok))
-		return
-	}
-}
-
-func (p *parser) parseVariable() Expression {
-	var value Expression
+		str, err := strconv.Unquote(p.scanner.TokenText())
+		if err != nil {
+			p.errorf("couldn't parse string: %s", err)
+			return
+		}
 
-	switch text := p.scanner.TokenText(); text {
-	case "true", "false":
-		value = &Bool{
+		value = &String{
 			LiteralPos: p.scanner.Position,
-			Value:      text == "true",
+			Value:      str,
 		}
-	default:
-		if p.eval {
-			if assignment, local := p.scope.Get(text); assignment == nil {
-				p.errorf("variable %q is not set", text)
-			} else {
-				if local {
-					assignment.Referenced = true
-				}
-				value = assignment.Value
+		p.accept(scanner.String)
+	case '-', scanner.Int: // Integer might have '-' sign ahead ('+' is only treated as operator currently.)
+		var str string
+		literalPos := p.scanner.Position
+		if p.tok == '-' {
+			if p.scanner.Peek() < 48 || p.scanner.Peek() > 57 {
```

Comment:
Could you replace "48" with " '0' " and " 57 " with " '9' " or otherwise make them into constants like perhaps numberZeroAsRune and numberNineAsRune?